### DISCUSSION TITLE
How to associate a record with a Auth user id?

### DISCUSSION THREAD

**Discussion by 0radek** [supabase/supabase#2166] [Questions]
Say I have a table `Posts` with a `title`, `message`, and `created_by`.

How should I set up my table so that I can reference the Auth user? I see my user has been created in the Auth table, but no way to see that table in my Posts table editor.

**Comment by silentworks** [comment]
If you are using the Supabase ui to create the tables, when you create the `created_by` column, you need to add foreign key relation and select the relating table along with its column.

<img width="254" alt="Screenshot 2021-07-01 at 00 36 36" src="https://user-images.githubusercontent.com/79497/124044089-19a28780-d9fc-11eb-9214-b57378350529.png">

Now when you enter new data along with its relating field, you should see a icon appear in the table editor next to the relating content.

<img width="48" alt="Screenshot 2021-07-01 at 00 38 10" src="https://user-images.githubusercontent.com/79497/124044187-5ec6b980-d9fc-11eb-8183-5d3c6801b7f3.png">

Clicking this icon should show you the relating table data.

**Reply by 0radek** [reply]
Thanks!

I also read this in the docs: https://supabase.io/docs/guides/auth#create-a-publicusers-table

What data type should I use for my primary key in order to do this: `create a users table in the public schema, which uses the same UID Primary Key as the auth.users`

does it mean to create a public.users table, and use int8 id, but also another column (like `suid`) that will have the supabase id?

**Comment by 0radek** [comment]
BTW the auth.users table doesn't show up in the "Add Foregin Key Relation" table select

**Reply by hallee9000** [reply]
I also want the auth user to show up here.

**Reply by chrisLoPresti** [reply]
not showing up for me, wonder if intentional

**Reply by hallee9000** [reply]
Hi @chrisLoPresti, it is intentional. See https://github.com/supabase/supabase/discussions/2345

**Reply by chrisLoPresti** [reply]
ty for the reply
